packages/pds/src/auth-verifier.ts

```diff
@@ -219,24 +229,39 @@ export class AuthVerifier {
     }
   }
 
-  adminService = async (reqCtx: ReqCtx): Promise<AdminServiceOutput> => {
-    const jwtStr = bearerTokenFromReq(reqCtx.req)
-    if (!jwtStr) {
-      throw new AuthRequiredError('missing jwt', 'MissingJwt')
-    }
-    const payload = await verifyServiceJwt(
-      jwtStr,
-      this.dids.entryway ?? this.dids.pds,
-      async (did, forceRefresh) => {
-        if (did !== this.dids.admin) {
-          throw new AuthRequiredError(
-            'Untrusted issuer for admin actions',
-            'UntrustedIss',
-          )
-        }
-        return this.idResolver.did.resolveAtprotoKey(did, forceRefresh)
+  userDidAuth = async (reqCtx: ReqCtx): Promise<UserDidOutput> => {
+    const payload = await this.verifyServiceJwt(reqCtx, {
+      aud: this.dids.entryway ?? this.dids.pds,
+      iss: null,
+    })
+    return {
+      credentials: {
+        type: 'user_did',
+        aud: payload.aud,
+        iss: payload.iss,
       },
-    )
+    }
+  }
+
+  userDidAuthOptional = async (
+    reqCtx: ReqCtx,
+  ): Promise<UserDidOutput | NullOutput> => {
+    if (isBearerToken(reqCtx.req)) {
+      try {
+        return await this.userDidAuth(reqCtx)
+      } catch {
+        return { credentials: null }
+      }
+    } else {
+      return { credentials: null }
+    }
+  }
+
+  adminService = async (reqCtx: ReqCtx): Promise<AdminServiceOutput> => {
+    const payload = await this.verifyServiceJwt(reqCtx, {
+      aud: this.dids.entryway ?? this.dids.pds,
+      iss: [this.dids.admin],
```

Comment:
In a world where pdses have dids, I wonder if we'll eventually want service auth to include the pds as the issuer and the user as the subject.  Not relevant to this PR, just wanted to preserve the thought before it slips away!

Reply:
yup I could totally see that 👍